src/borg/helpers/fs.py

```diff
@@ -164,12 +164,39 @@ def dir_is_tagged(path, exclude_caches, exclude_if_present):
     return tag_names
 
 
-_safe_re = re.compile(r"^((\.\.)?/+)+")
+def make_path_safe(path):
+    """
+    Make path safe by making it relative and normalized
 
+    `path` is normalized by making it relative, removing
+    consecutive slashes (e.g. '//'), remove '.' elements,
+    and removing trailing slashes.
 
-def make_path_safe(path):
-    """Make path safe by making it relative and local"""
-    return _safe_re.sub("", path) or "."
+    For reasons of security, a ValueError is raised should
+    `path` contain any '..' elements.
+    """
+    path = path.lstrip("/")
+    if path.startswith("../") or "/../" in path or path.endswith("/..") or path == "..":
+        raise ValueError(f"unexpected '..' element in path {path!r}")
+    path = os.path.normpath(path)
```

Comment:
https://github.com/borgbackup/borg/issues/7120 i filed this because we use normpath quite a lot.

as this has this separate issue now, we can ignore it for the scope of this PR.